Add code of conduct and redirects to our docs

jhkennedy opened the pull request:
The two redirects should ensure that these links get redirected appropriately:
* `https://https://earthaccess.readthedocs.io/CODE_OF_CONDUCT`
* `https://https://earthaccess.readthedocs.io/CODE_OF_CONDUCT.md`

Had to add mkdocs-redirects (I've used it a lot; works great) and so the pyproject.toml got updated and the lock file.

fixes #591 


<!-- readthedocs-preview earthaccess start -->
----
📚 Documentation preview 📚: https://earthaccess--592.org.readthedocs.build/en/592/

<!-- readthedocs-preview earthaccess end -->

jhkennedy commented:
IDK why mkdocs isn't building -- worked fine on my machine, but likely a dependency issue from adding `mkdocs-redirects` and poetry being poetry.

chuckwondo commented:
> IDK why mkdocs isn't building -- worked fine on my machine, but likely a dependency issue from adding `mkdocs-redirects` and poetry being poetry.

If I can manage to get this PR merged, it includes a fix for the mkdocs build failure: https://github.com/nsidc/earthaccess/pull/571 (@mfisher87, nudge, nudge)

mfisher87 commented:
Good to go! :snail: :rocket:

chuckwondo commented:
@jhkennedy, if you want to rebase/merge main into your branch, your build should pass now.

jhkennedy commented:
@chuckwondo @mfisher87 everything's passing now.

Notably, I added `mkdocs-redirects` to the dev dependencies. For the rebase, I resolved conflicts in the `pyproject.toml` and updated the poetry lock file like  `poetry lock --no-update`. 

I also don't remember/understand what's going on with:
https://github.com/nsidc/earthaccess/blob/main/ci/environment-dev.yml

I see most of the mkdocs stuff there, but I don't really understand why we have things there beyond just:
```yaml
name: earthaccess-dev
channels:
  - conda-forge
dependencies:
  - python=3.10
  - poetry
```
and I'm not sure we even need to pin Python here, either.

But anyway, for this PR, do I need to update `envionment-dev.yml` to include `mkdocs-redirects`?

mfisher87 commented:
I would like to better understand the reason for the environment-dev.yml file as well. A lot of overhead to update it. I think we should add the new dependency there for now, and maybe let's make a new issue to discuss removing the file?

jhkennedy commented:
lol, docs timed out in the notebook execution step. 

I added it mostly to re-trigger the docs build, but the only different between it being in the env vs not is whether it's installed from conda-forge or pypi (doesn't matter here).

I'll open an issue for the `environment-dev.yml` file.

mfisher87 commented:
>lol, docs timed out in the notebook execution step.

This happened on another PR too. I think RTD had a hickup.

mfisher87 commented on docs/contributing/code-of-conduct.md:
Works great!

mfisher87 commented on docs/contributing/index.md:
```diff
@@ -6,7 +6,7 @@ with the community and maintainers via
 or [any other method](our-meet-ups.md).
 
-Please note that we have a [code of conduct](/CODE_OF_CONDUCT.md). Please follow it in all of your interactions with the project.
+Please note that we have a [code of conduct](./code-of-conduct.md). Please follow it in all of your interactions with the project.
```
wonder why strict mode didn't catch this?

jhkennedy replied:
@mfisher87 Absolute links aren't officially supported on MkDocs because they can be unpredictable and are not validated because they are effectively external links.
https://www.mkdocs.org/user-guide/writing-your-docs/#linking-to-pages

For example, if you host multiple things on a domain, this is a way to link them without knowing the explicit domain name, e.g.:

* root: <pacakge.com>
* MkDocs site URL: <package.com/docs/>
* download artifacts: <example.com/downloads/>

So `/downloads/package-linux.tar.gz` would be a valid link for a tarball you might want to document on the site.


This is probably a good policy as it's really hard to say what `/` is intended to be --  here, the absolute link was intended to be back to the *repository root*, not to the docs dir or the domain root.

mfisher87 reviewed (approved):
LGTM! @jhkennedy do you want to merge this when you're ready?

jhkennedy commented:
> LGTM! @jhkennedy do you want to merge this when you're ready?

Merged! @mfisher87 is that sufficient for this to be deployed to readthedocs, or do we need a release as well?

mfisher87 commented:
This is deployed on ReadTheDocs (https://readthedocs.org/projects/earthaccess/builds/24666108/), but unfortunately, it doesn't seem to have resolved our issue. It's not an Mkdocs issue, it's a ReadTheDocs issue.

I set up an "exact redirect" in the admin dashboard following [the RTD redirects documentation](https://docs.readthedocs.io/en/stable/user-defined-redirects.html#user-defined-redirects) and now the link in the proposal works.

mfisher87 commented:
@jhkennedy should we revert this PR since the redirect is handled by RTD? The reason we can't solve this in Mkdocs is because of RTD handles the language and version URL parts before hitting the Mkdocs output.
